bears/general/ShebangBear.py

```diff
@@ -0,0 +1,33 @@
+import os
+from coalib.bears.LocalBear import LocalBear
+from coalib.results.Diff import Diff
+from coalib.results.Result import Result
+
+
+class ShebangBear(LocalBear):
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_FIX = {'Syntax'}
+
+    def run(self, filename, file):
+        """
+        Ensure that the file uses the generic Shebang operator.
+        """
+        if not file or file[0].startswith('#!/usr/bin/env'):
+            return
+        if file[0].startswith('#!'):
+            diff = Diff(file)
+            message = None
+            binary = os.path.basename(file[0]).strip('\n')
```

Comment:
don't you have to strip off the `#!` part in `file[0]`?
So in the end you have `file[0][2:]`?

Reply:
I don't know if i understood your comment but the basename extracts the binary file out of the path.
https://docs.python.org/2/library/os.path.html#os.path.basename
Why would i strip the #! if i don't care what the first part of the path is?

Reply:
but is `os.path.basename` able to handle it? As the path starts after `#!`

Reply:
https://github.com/python/cpython/blob/3.6/Lib/posixpath.py#L142
The method just extracts all the characters after the last separator.
IMO, we shouldn't strip off the `#!`.
What do you think?

Reply:
Those characters don't belong to the path, that's why I'm asking for it, as conceptually it's not "completely right". I would do `file[0][2:]` for the sake of comprehensibility ;)